ImPlot Performance Improvement

calvertdw opened the pull request:
I would like to make the methods public to enable use of primitive arrays. Otherwise, to use the boxed types, we are creating whole new arrays every frame, which is terrible performance wise as it contributes to frequent garbage collection and incurs extra data copying, using up CPU.

@perrymacmurray Was there a reason they are private or why we should not pass primitive arrays directly?

perrymacmurray commented:
I don't think there's a particular reason (or at least, a good reason) for `nPlotLine` method being private, I just looked at the method again and I think it was only to match the other native methods. I would consider changing the name of `nPlotLine` to an overload of `plotLine`, though, just to match the original C++ ImPlot API, though.

calvertdw commented:
I made some separate ones that are public and dropped the n* prefix. I also removed the warning about information loss because that shouldn't be an issue for these. I went ahead and just made the API passthrough instead of trying to be smart about the size, just for performance possibilities.

calvertdw commented:
Okay if it passes I think I'm all done.
